app/models/game.rb

```diff
@@ -99,4 +88,89 @@ def next_turn!
       self.update_attributes(current_turn: "white")
     end
   end
+
+#end game check and checkmate
+  def check?
+    kings = King.where(game_id: id)
+    kings.each do |king|
+      pieces.each do |piece|
+        return true if piece.color != king.color && piece.valid_move?(king.x_cord, king.y_cord)
+      end
+    end
+
+    return false
+  end
+
+  def checkmate?
+    return false if method1? || method2? || method3?
```

Comment:
great job separating the 3 different logical paths... now lets name them 😉  using your comments we can give them names based on what they check

return false if king_can_move_out_of_check? || ... || ...